Automattic/wp-calypso
# Jetpack Manage: Pricing page - implement non logged in redirect flow to issue license page.

Related to https://github.com/Automattic/jetpack-manage/issues/259

## Proposed Changes

* This PR implements a flow that leads non-logged in users to the `partner-portal/issue-license` page.
* If not a partner, they will land on the `partner-portal/partner` page, and can then click the 'Sign Up' button. Clicking on this will take you to the the Manage signup page. Filling out the form on this page will then result in a redirect to the issue licenses page.
* If you are a partner you should be redirected straight to the issue licenses page
* On the `partner-portal/partner` page (where you land after logging in when you are not a partner), Initially I was going to hide the new 'sign up' button behind a check for the relevant URL queryParams, but it seems relevant for anyone coming to that page (as does the paragraph text linking to the `manage` page).
* This PR also takes the logic out of the plan and product card files and creates a new component that creates the relevant links.

## Testing Instructions


To test, via the `http://jetpack.cloud.localhost:3000/manage/pricing` page. On any product or plan card:
* The Get buttons as a logged in partner should take you the issue licenses page (note that you may need to refresh the page as there is an existing issue resulting in an error message on first visit, in some cases).
* The Get button as a logged in non-partner should take you to the manage sign up page. You can sign up to confirm that you are redirected to the issue licenses page.
* The Get button for a non logged in partner will take you to the WPcom login page, then to the issue licenses page.
* The Get button for a non logged-in non-partner will take you to the WPCom login page then the partner-portal 'Licensing' page. There is a new button there to sign up for Jetpack Manage (and new paragraph text linking to the /manage/ info page). Click that, and you'l be taken to the sign up page - then redirected from there to the issue licenses page once signed up.

Note that to test the non-logged in route while in development, you may need to directly use a link as would be clicked on the `manage/pricing` page: `http://jetpack.cloud.localhost:3000/partner-portal/issue-license/?product_slug=jetpack-anti-spam&source=manage-pricing-page`.
To replicate visiting the `partner-portal/partner` page without going through the login process (as a non partner): `http://jetpack.cloud.localhost:3000/partner-portal/partner?return=%2Fpartner-portal%2Fissue-license%3Fproduct_slug%3Djetpack-anti-spam%26source%3Dmanage-pricing-page` (though it would be preferable to test the real flow).


Screenshot for the `partner-portal/partner` page when not a partner:

<img width="567" alt="Screenshot 2024-01-31 at 11 14 49" src="https://github.com/Automattic/wp-calypso/assets/16754605/18768f22-95d9-46b4-aa61-ccca976f82ae">

## Pre-merge Checklist
